### PR TITLE
Update Recipient Signature generation reference code in python

### PR DESCRIPTION
**Pull request by LuZhang-Lou**
I've verified this snippet of code against testnet

### CODE REVIEW

**Comment by n4ss** [comment]
I think it will be easier to read if we keep it very explicit. wdyt of:
```
from pylibra import lcs
from pylibra import libra_types as libra
from pylibra import serde_types as st
...

metadata: Metadata = libra.Metadata__TravelRuleMetadata(
  libra.TravelRuleMetadata__TravelRuleMetadataVersion0(
    libra.TravelRuleMetadataV0(
      off_chain_reference_id="reference_id",
    )
  )
)
// `lcs_metadata` will be used as the `metadata` parameter in encode_peer_to_peer_with_metadata_script()
lcs_metadata = lcs.serialize(metadata, libra.Metadata)

dual_attest_msg_bytes: bytearray = bytearray()
dual_attest_msg_bytes.extend(lcs_metadata)  # Travel Rule metadata
dual_attest_msg_bytes.extend(lcs.serialize(bytes.fromhex("793251de1a61e9b4d1a17d13aa015e45"), bytes))  # Raw on-chain address (https://github.com/libra/lip/blob/master/lips/lip-5.md#terminology)
dual_attest_msg_bytes.extend(lcs.serialize(st.uint64(1000000), st.uint64))  # Amount
dual_attest_msg_bytes.extend(b"@@$$LIBRA_ATTEST$$@@")  # ASCII-encoded Libra Domain Separator string

// We sign the bytes directly with the compliance private key
// `receiver_metadata_signature` will be used as the `metadata_signature` parameter in  encode_peer_to_peer_with_metadata_script()
receiver_metadata_signature = compliance_private_key.sign(dual_attest_msg_bytes)
```

**Comment by LuZhang-Lou** [comment]
> I think it will be easier to read if we keep it very explicit. wdyt of:
> 
> ```
> from pylibra import lcs
> from pylibra import libra_types as libra
> from pylibra import serde_types as st
> ...
> 
> metadata: Metadata = libra.Metadata__TravelRuleMetadata(
>   libra.TravelRuleMetadata__TravelRuleMetadataVersion0(
>     libra.TravelRuleMetadataV0(
>       off_chain_reference_id="reference_id",
>     )
>   )
> )
> // `lcs_metadata` will be used as the `metadata` parameter in encode_peer_to_peer_with_metadata_script()
> lcs_metadata = lcs.serialize(metadata, libra.Metadata)
> 
> dual_attest_msg_bytes: bytearray = bytearray()
> dual_attest_msg_bytes.extend(lcs_metadata)  # Travel Rule metadata
> dual_attest_msg_bytes.extend(lcs.serialize(bytes.fromhex("793251de1a61e9b4d1a17d13aa015e45"), bytes))  # Raw on-chain address (https://github.com/libra/lip/blob/master/lips/lip-5.md#terminology)
> dual_attest_msg_bytes.extend(lcs.serialize(st.uint64(1000000), st.uint64))  # Amount
> dual_attest_msg_bytes.extend(b"@@$$LIBRA_ATTEST$$@@")  # ASCII-encoded Libra Domain Separator string
> 
> // We sign the bytes directly with the compliance private key
> // `receiver_metadata_signature` will be used as the `metadata_signature` parameter in  encode_peer_to_peer_with_metadata_script()
> receiver_metadata_signature = compliance_private_key.sign(dual_attest_msg_bytes)
> ```


Seems like the only difference of this and the original code is DOMAIN_SEPARATOR is not serialized? I tried this and it did not work. Did you test this snippet in testnet ? @n4ss

**Comment by n4ss** [comment]
Domain separator is not serialized and the byte vector is not serialized either. This is the same as your implem

**Comment by LuZhang-Lou** [comment]
`lcs.serialize(bytes.fromhex("793251de1a61e9b4d1a17d13aa015e45"), bytes) ` is not the correct serialization for address. I think it's because the address has a different type than normal bytes.  You can get the lcs bytes by calling `utils.account_address(addr_hex).lcs_serialize()`

and If your point is dataclass is less clear to read, I've updated to also include a version that concats individually serialized bytes (very similar to the original code snippet before this change)

File: lips/lip-1.mdx [hunk 1]
```diff
@@ -544,20 +533,40 @@ metadata: libra.Metadata = libra.Metadata__TravelRuleMetadata(
...
-lcs_metadata = lcs.serialize(metadata, libra.Metadata)
+
+# Method2: concat individually serialized bytes
+
+dual_attest_bytearray: bytearray = bytearray()
+dual_attest_bytearray.extend(lcs_metadata)
+dual_attest_bytearray.extend(utils.account_address(sender_libra_address_hex).lcs_serialize())
+dual_attest_bytearray.extend(lcs.serialize(st.uint64(amount), st.uint64))
+dual_attest_bytearray.extend(DOMAIN_SEP)
+dual_attest_msg_lcs = bytes(dual_attest_bytearray)
```
Comment (n4ss) [inline lips/lip-1.mdx:569]: nit: dual_attest_msg_lcs this is not lcs encoded anymore

**Review by n4ss — APPROVED** [review]

File: lips/lip-1.mdx [hunk 2]
```diff
@@ -515,29 +515,59 @@ In the event of a soft match, the VASP associated with this actor will need to d
...
+metadata: libra.Metadata = libra.Metadata__TravelRuleMetadata(
   libra.TravelRuleMetadata__TravelRuleMetadataVersion0(
     libra.TravelRuleMetadataV0(
       off_chain_reference_id="reference_id",
     )
   )
 )
+
 // `lcs_metadata` will be used as the `metadata` parameter in encode_peer_to_peer_with_metadata_script()
 lcs_metadata = lcs.serialize(metadata, libra.Metadata)
```
Comment (kphfb) [inline lips/lip-1.mdx:537]: Looks like there is an extra space leading this line, can you remove it?

File: lips/lip-1.mdx [hunk 3]
```diff
@@ -515,29 +515,59 @@ In the event of a soft match, the VASP associated with this actor will need to d
...
+    amount=st.uint64(amount),
+)
+dual_attest_msg_lcs = attest.lcs_serialize() + DOMAIN_SEP
+
+# Method2: concat individually serialized bytes
 
-dual_attest_msg = libra.TransactionArgument__U8Vector(bytes(dual_attest_msg_bytes))
-dual_attest_msg_lcs = lcs.serialize(dual_attest_msg, libra.TransactionArgument__U8Vector)
+dual_attest_bytearray: bytearray = bytearray()
+dual_attest_bytearray.extend(lcs_metadata)
```
Comment (kphfb) [inline lips/lip-1.mdx:565]: since lcs_metadata is only used for method2, let's move its definition down here also

File: lips/lip-1.mdx [hunk 4]
```diff
@@ -515,29 +515,59 @@ In the event of a soft match, the VASP associated with this actor will need to d
...
 // `lcs_metadata` will be used as the `metadata` parameter in encode_peer_to_peer_with_metadata_script()
 lcs_metadata = lcs.serialize(metadata, libra.Metadata)
 
-dual_attest_msg_bytes: bytearray = bytearray()
-dual_attest_msg_bytes.extend(lcs_metadata)  # Travel Rule metadata
-dual_attest_msg_bytes.extend(lcs.serialize(bytes.fromhex("793251de1a61e9b4d1a17d13aa015e45"), bytes))  # Raw on-chain address (https://github.com/libra/lip/blob/master/lips/lip-5.md#terminology)
-dual_attest_msg_bytes.extend(lcs.serialize(st.uint64(1000000), st.uint64))  # Amount
-dual_attest_msg_bytes.extend(lcs.serialize(b"@@$$LIBRA_ATTEST$$@@", bytes))  # ASCII-encoded Libra Domain Separator string
+sender_libra_address_hex = "793251de1a61e9b4d1a17d13aa015e45
+sender_address = utils.account_address(sender_libra_address_hex)
```
Comment (kphfb) [inline lips/lip-1.mdx:540]: This is only used for method1, so let's move it into that block

File: lips/lip-1.mdx (hunk 1 above)
Comment (kphfb) [reply lips/lip-1.mdx:569]: Yeah, let's rename it to just dual_attest_msg

Comment (kphfb) [reply lips/lip-1.mdx:569]: Both here and in example1

Comment (kphfb) [reply lips/lip-1.mdx:569]: And in the receiver_metadata_signature line below

**Review by n4ss — APPROVED** [review]
👍📈

**Review by dahliamalkhi — APPROVED** [review]

**Pull request merged**